app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -142,4 +142,15 @@ def available_vlans_and_hosts(options = {})
     end
     return vlans, hosts
   end
+
+  def allowed_storage_profiles(options = {})
+    return [] if (src = resources_for_ui).blank? || src[:vm].nil?
+    model_name = options[:storage_profile_filter]
+    return @filters[model_name] unless @filters[model_name].nil?
```

Comment:
You can remove some duplication of `@filters[model_name]` by using:

```ruby
@filters['StorageProfile'] ||= begin
  template = load_ar_obj(src[:vm])
  ...
  storage_profiles.each_with_object({}) { |s, m| m[s.id] = s.name }
end
```

Then you do not even need `model_name` anymore.